File: rb/features.rb
# Unirate SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module UnirateFeatures
  def self.make_feature(name)
    case name
    when "base"
      UnirateBaseFeature.new
    when "ratelimit"
      UnirateRatelimitFeature.new
    when "retry"
      UnirateRetryFeature.new
    when "test"
      UnirateTestFeature.new
    when "timeout"
      UnirateTimeoutFeature.new
    else
      UnirateBaseFeature.new
    end
  end
end

File: py/features.py
# Unirate SDK feature factory

from feature.base_feature import UnirateBaseFeature
from feature.test_feature import UnirateTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UnirateBaseFeature(),
        "test": lambda: UnirateTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
